pkg/install/generated_templates.gogen.go

```diff
@@ -19,39 +19,39 @@ var templates = func() http.FileSystem {
 	fs := vfsgen۰FS{
 		"/": &vfsgen۰DirInfo{
 			name:    "/",
-			modTime: time.Date(1970, 1, 1, 0, 0, 0, 0, time.UTC),
+			modTime: time.Date(2019, 10, 17, 13, 27, 25, 101756322, time.UTC),
```

Comment:
Is this on purpose or did you forget to restore the `modTimeFS` wrapper?

Reply:
Honestly I was hoping it wasn't needed any more. Since check-generated will only fail if the instantiated templates are different to those in deploy/, there's no need to commit meaningless changes to the generated code (i.e., just because a template got a fresh mtime).

Thus ugly conflicts will only arise if there are genuinely changes on both branches -- and those should be resolvable just by making sure the non-generated files are merged OK, then regenerating this file.

Does this follow, or have I misunderstood the motivation of modTimeFS?

Reply:
Actually, I can check, since you outlined how this happens in Slack:

> 1. work on some branch that is older than master
> 2. rebase this branch, on master, the timestamp of your files is now newer than master
> 3. switch back to master, because git does not see any changes (they are equal by content to what's there), git will not touch them
> 4. profit, your timestamp is now newer than master and check-generated will fail

Reply:
I can cause problems by doing this:

 1. from a `base` branch, create branch `test`
 2. commit a change to a template in `test`
 3. run `make check-generated`, observe that it fails (as it should)
 4. commit the regenerated Go source in pkg/install/, and the generated yamls in deploy/
 5. checkout `base`, make any commit, then checkout `test` again -- this will touch the template you changed, since it will be replaced each time you change branch
 6. at this point, `make check-generated` will succeed (it produces the same deploy/ yaml that is committed); but it will regenerate the Go code, with different timestamps.
 7. rebasing on `base`, or similar operations, will probably break because of the spuriously regenerated Go file.

So it's not a disastrous problem -- everything will still build, and you can get past it (just overwrite the Go file) if you understand what's going on -- but it is a bit of a trap, and as with the origin problem, will come up frequently.

Reply:
> but it is a bit of a trap, and as with the origin problem, will come up frequently.

I think this is enough reason to put it back, as we became tired of the issue pretty soon last time.

Reply:
> I think this is enough reason to put it back

Me also too